docs/docs/guides/multi-stage-delivery.md

```diff
@@ -0,0 +1,258 @@
+- Followed the [instructions for setting up observability](./otel.md#requirements-for-opentelemetry)
+- Access to a GitHub repository for hosting your application manifests.
+- A Personal Access token for your GitHub repository.
+We recommend using a [fine-grained token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#fine-grained-personal-access-tokens)
+with the following permissions for the repository:
+    - Read access to metadata
+    - Read/Write access to actions
+
+Note that for the promotion to work properly,
+you need to enable read and write permissions for
+Github actions in your repository - see the screenshot below:
+
+![Workflow Permissions](./assets/multi-stage-delivery/workflow-permissions.png)
+
+## Preparing the Application Manifests
+
+In this example, the application will be deployed in
+two different namespaces, each representing a different
+stage (`dev` and `production`).
+To create the namespaces, execute the following commands:
+
+```shell
+kubectl create namespace podtato-head-dev
+kubectl annotate namespace podtato-head-dev keptn.sh/lifecycle-toolkit=enabled
+kubectl create namespace podtato-head-production
+kubectl annotate namespace podtato-head-production keptn.sh/lifecycle-toolkit=enabled
+```
+
+The promotion task that triggers the action to
+create a pull request for promoting an application version
+from `dev` to `production` will be executed in the `podtato-head-dev` stage.
+Therefore, we need to create a secret containing the GitHub personal
+access token in this namespace, using the following command:
+
+```shell
+GH_REPO_OWNER=<YOUR_GITHUB_USER>
+GH_REPO=<YOUR_GITHUB_REPO>
+GH_API_TOKEN=<YOUR_GITHUB_TOKEN>
+kubectl create secret generic github-token -n podtato-head-dev --from-literal=SECURE_DATA="{\"githubRepo\":\"${GH_REPO}\",\"githubRepoOwner\":\"${GH_REPO_OWNER}\",\"apiToken\":\"${GH_API_TOKEN}\"}"
+```
+
+Next, add the promotion task as a GitHub action by
+committing the following file to the repository:
+
+```yaml title=".github/workflows/promote.yaml"
+{% include "./assets/multi-stage-delivery/promote.yaml" %}
+```
+
+Next, add the application manifests to the GitOps repository.
+Note that in this example, the manifests for the `dev` and `production`
+stage are located in two different subdirectories within
+the repository to keep it simple.
+This might differ heavily depending on how your organization's
+GitOps repositories and application stages are organized.
+
+Inside your git repository, create the following files
+containing the manifests for the `dev` and `production` stage:
+
+**Dev Stage:**
+
+<!-- markdownlint-disable MD046 -->
+
+=== "dev/templates/application.yaml"
+
+    ```yaml
+    {% include "./assets/multi-stage-delivery/application-dev.yaml" %}
+    ```
+
+=== "dev/Chart.yaml"
+
+    ```yaml
+    {% include "./assets/multi-stage-delivery/chart.yaml" %}
+    ```
+
+=== "dev/values.yaml"
```

Comment:
should we add oneliner explaination for each why they are needed? I know it is redundant but for instance to help user understand promotion magic example works due to the values file